### PR TITLE
HW 5 - Architecture

### PR DESCRIPTION
**Pull request by Markusplay**
# Layers:

<ul>
    <li>View</li>
    <li>Application</li>
    <li>Infrastructure</li>
</ul>

### Sublayers:
<h3><b>Presentation</b></h3> 
It represents ui components that doesn't do lot of business logic and doesn't work directly with Api, store and Infrastructure player.

- CourseCard
- CoursesPage
- ListLessons
<h3><b>Container</b></h3> 
It knows and communicates with Application layer, Api and store.

- Main Page
- Course Page
- Not Found Page

## The Application layer

<h3><b>Hooks</b></h3> 
They know about Infrastructure layer and can work with it.

- useSelect

## The Infrastructure layer.
It is responsible for requests to the server, manage and save data.

<h3><b>Service/api/instance.ts</b></h3>
This is used solely for making API requests to a backend server. In this case, it provides a level of abstraction between the application and the server, allowing the application to make requests without having to worry about low-level details like handling HTTP status codes and formatting request data.

<h3><b>Redux</b></h3>
In this project is also used Redux as state management system. 
For corrcet work of the redux we should implement:

* asynActions
* slices
* store
* selectors

# Schema

arch

### CODE REVIEW

File: src/pages/course-page/CoursePage.tsx [hunk 1]
```diff
@@ -73,44 +76,23 @@ const CoursePage = () => {
...
-          <h4 className={styles.content}>Lessons:</h4>
+          <h2 className={styles.content}>Lessons:</h2>
           {sortedLessons?.map(lesson => (
             <ListLessons
-              duration={lesson.duration}
-              setCurrentLesson={setCurrentLesson}
-              order={lesson.order}
               key={lesson.id}
-              title={lesson.title}
+              lesson={lesson}
```
Comment (yulyaolshanska) [inline src/pages/course-page/CoursePage.tsx]: Why did you decide to transfer not only the necessary data (duration, order, title), but the entire lesson?

File: src/utils/sortCourses.tsx [hunk 2]
```diff
@@ -0,0 +1,12 @@
+import { Course } from '../redux/type';
+
+export const sortCourses = (courses: Course[] | undefined): Course[] => {
+  return (
+    courses
+      ?.slice()
+      .sort(
+        (a: any, b: any) =>
```
Comment (yulyaolshanska) [inline src/utils/sortCourses.tsx]: any?

Comment (Markusplay) [reply src/utils/sortCourses.tsx]: deprecated file, it was fixed

File: src/pages/course-page/CoursePage.tsx (hunk 1 above)
Comment (Markusplay) [reply src/pages/course-page/CoursePage.tsx]: > Why did you decide to transfer not only the necessary data (duration, order, title), but the entire lesson?

I decided to put all this things that came from lesson of Course Page in LessonList as it is and then destructuring all properties I need in component. + It is easier to read code in Course Page from the point of semantic.

**Review by yulyaolshanska — APPROVED** [review]

**Pull request merged**